controllers/space/space_controller_test.go

```diff
@@ -1794,6 +1794,219 @@ func TestRetargetSpace(t *testing.T) {
 	})
 }
 
+// TestSubSpace covers the cases where there is a relationship between a parentSpace and a subSpace
+// and all the consequences and expectations of this link between two different spaces.
+func TestSubSpace(t *testing.T) {
+
+	// given
+	logf.SetLogger(zap.New(zap.UseDevMode(true)))
+	s := scheme.Scheme
+	err := apis.AddToScheme(s)
+	require.NoError(t, err)
+	basicTier := tiertest.BasicTier(t, tiertest.CurrentBasicTemplates)
+
+	// test SpaceBindings of a parentSpace are created/updated/deleted ,
+	// and how this should affect the subSpace and its NSTemplateSet
+	t.Run("SpaceBindings inheritance ", func(t *testing.T) {
+
+		t.Run("create parentSpace with admin and viewer roles, and expect subSpace will have same usernames and roles", func(t *testing.T) {
+			// given a Space and its NSTemplateSet resource...
+			parentSpace := spacetest.NewSpace("parentSpace",
+				spacetest.WithTierName(basicTier.Name),
+				spacetest.WithSpecTargetCluster("member-1"),
+				spacetest.WithStatusTargetCluster("member-1"), // already provisioned on a target cluster
+				spacetest.WithFinalizer())
+			parentNStmplSet := nstemplatetsettest.NewNSTemplateSet(parentSpace.Name,
+				nstemplatetsettest.WithReferencesFor(basicTier,
+					// include pre-existing users with role...
+					nstemplatetsettest.WithSpaceRole("admin", "parentAdmin"),
+					nstemplatetsettest.WithSpaceRole("viewer", "parentViewer"),
+				),
+				nstemplatetsettest.WithReadyCondition(),
+			)
```

Comment:
for the purpose of this test, you don't have to define all the things for parent-Space CR - this resource is not being reconciled, so having just the resource present should be enough. 
In other words, for the simplicity of the test I would drop the extra part:
```suggestion
			// given a Space...
			parentSpace := spacetest.NewSpace("parentSpace")
```
The same applies to the following tests as well.